src/Servers/Kestrel/Core/src/Internal/Http/Http1MessageBody.cs

```diff
@@ -72,7 +72,7 @@ protected override Task OnConsumeAsync()
             _context.ReportApplicationError(connectionAbortedException);
 
             // Have to abort the connection because we can't finish draining the request
-            _context.StopProcessingNextRequest();
+            _context.StopProcessingNextRequest(ConnectionEndReason.AbortedByApplication);
```

Comment:
It feels like we might want to distinguish between "the application requested an abort" and "the application failed and we had to abort".

Reply:
I went with `AbortedByApp` because the log message says the app aborted the connection. I agree that the reason isn't clear.

What do you think about the logging and end reason here @halter73 @Tratcher @mgravell ?

Reply:
I think the assumption here is that the only reason `TryReadInternal` would throw an `InvalidOperationException` is because middleware exited after calling `HttpContext.Request.BodyReader.ReadAsync` without a corresponding call to `AdvanceTo`, and rather than try to recover we bail out since we assume it's an unlikely scenario anyway.

I should have created a custom message explaining this rather than reuse `CoreStrings.ConnectionAbortedByApplication`  for the exception passed to `ReportApplicationError`.

Ideally, we'd use "application_abort" only for `HttpContext.Abort()` on HTTP/1.1 connections, and then use "application_left_bodyreader_in_invalid_state" for this, and "application_canceled_response_flush" for the usage of `AbortedByApplication` in `TimingPipeFlusher`.

If we reused the same `ConnectionEndReason` for `HttpCotnext.Abort()` and this issue with leaving the body reader in an invalid state, I won't be too upset since this scenario is pretty niche. Cancelling a response body flush on an HTTP/1.1 connection probably deserves its own `ConnectionEndReason` though.